netobserv/v1alpha1/flowcollector_types.go

```diff
@@ -0,0 +1,436 @@
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	AgentIPFIX = "ipfix"
+	AgentEBPF  = "ebpf"
+)
+
+// Please notice that the FlowCollectorSpec's properties MUST redefine one of the default
+// values to force the definition of the section when it is not provided by the manifest.
+// This will cause that the remaining default fields will be set according to their definition.
+// Otherwise, omitting the sections in the manifest would lead to zero-valued properties.
+// This is a workaround for the related issue:
+// https://github.com/kubernetes-sigs/controller-tools/issues/622
+
+// FlowCollectorSpec defines the desired state of FlowCollector
+type FlowCollectorSpec struct {
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//+kubebuilder:default:=""
+	// Namespace where NetObserv pods are deployed.
+	// If empty, the namespace of the operator is going to be used.
+	Namespace string `json:"namespace,omitempty"`
+
+	//+kubebuilder:validation:Enum=ipfix;ebpf
+	//+kubebuilder:default:=ipfix
+	// Select the flows tracing agent. Possible values are "ipfix" (default) to use
+	// the IPFIX collector, or "ebpf" to use NetObserv eBPF agent. When using IPFIX with OVN-Kubernetes
+	// CNI, NetObserv will configure OVN's IPFIX exporter. Other CNIs are not supported, they could
+	// work but necessitate manual configuration.
+	Agent string `json:"agent"`
+
+	// Settings related to IPFIX-based flow reporter when the "agent" property is set
+	// to "ipfix".
+	// +kubebuilder:default:={sampling:400}
+	IPFIX FlowCollectorIPFIX `json:"ipfix,omitempty"`
+
+	// Settings related to eBPF-based flow reporter when the "agent" property is set
+	// to "ebpf".
+	// +kubebuilder:default={imagePullPolicy:"IfNotPresent"}
+	EBPF FlowCollectorEBPF `json:"ebpf,omitempty"`
+
+	// Settings related to the flowlogs-pipeline component, which collects and enriches the flows, and produces metrics.
+	FlowlogsPipeline FlowCollectorFLP `json:"flowlogsPipeline,omitempty"`
+
+	// Settings related to the Loki client, used as a flow store.
+	Loki FlowCollectorLoki `json:"loki,omitempty"`
+
+	// Kafka configuration, allowing to use Kafka as a broker as part of the flow collection pipeline.
+	// This is a new and experimental feature, not yet recommended to use in production.
+	// +optional
+	Kafka FlowCollectorKafka `json:"kafka,omitempty"`
+
+	// Settings related to the OpenShift Console plugin, when available.
+	ConsolePlugin FlowCollectorConsolePlugin `json:"consolePlugin,omitempty"`
+
+	// Settings related to the OpenShift Cluster Network Operator, when available.
+	ClusterNetworkOperator ClusterNetworkOperatorConfig `json:"clusterNetworkOperator,omitempty"`
+
+	// Settings related to OVN-Kubernetes CNI, when available. This configuration is used when using OVN's IPFIX exports, without OpenShift. When using OpenShift, refer to the `clusterNetworkOperator` property instead.
```

Comment:
This feels like it might want to be part of the union as well.

We use unions to tell users "this piece of config is a choose one", and it makes it easier to say, ok, in this specific scenario, you can only set these options, rather than having it documented here like you have